Add documentation for `kubernetes_state_core` core check (#12552)
DataDog/integrations-core: kubernetes_state_core/README.md

# Agent Check: Kubernetes State Core

## Overview

Get metrics from Kubernetes service in real-time to:

- Visualize and monitor Kubernetes states.
- Be notified about Kubernetes failovers and events.

The Kubernetes State Metrics Core check leverages [kube-state-metrics version 2+][1] and includes major performance and tagging improvements compared to the legacy `kubernetes_state` check.

As opposed to the legacy check, with the Kubernetes State Metrics Core check, you no longer need to deploy `kube-state-metrics` in your cluster.

Kubernetes State Metrics Core provides a better alternative to the legacy `kubernetes_state` check as it offers more granular metrics and tags. See the [Major Changes][2] and [Data Collected][3] for more details.

## Setup

### Installation

The Kubernetes State Metrics Core check is included in the [Datadog Cluster Agent][4] image, so you don't need to install anything else on your Kubernetes servers.

### Requirements

- Datadog Cluster Agent v1.12+

### Configuration

{{< tabs >}}
{{% tab "Helm" %}}

In your Helm `values.yaml`, add the following:

```yaml
datadog:
  # (...)
  kubeStateMetricsCore:
    enabled: true

> ```

Reviewer: Nit: KSM Core is enabled by default now in the v3+ of the helm chart

# Agent Check: Kubernetes State Core

## Overview

> Get Kubernetes metrics from [`kube-state-metrics`](https://github.com/kubernetes/kube-state-metrics) in real time to:

Reviewer 2: Could you add this link at the bottom of the file with the other links please? Thanks!

> Get Kubernetes metrics from [`kube-state-metrics`](https://github.com/kubernetes/kube-state-metrics) in real time to:

Reviewer 3: ```suggestion
Get Kubernetes metrics from [`kube-state-metrics`][1] in real time to:
```

> - Visualize and monitor Kubernetes states

Reviewer 2: ```suggestion
- Visualize and monitor Kubernetes states.
```

- Be notified about Kubernetes failovers and events.

This check supersedes the legacy `kubernetes_state` check.
Unlike the legacy check, it doesn't require any stand-alone `kube-state-metrics` deployment.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

### Installation

The Kubernetes State Core check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

1. Edit the `kubernetes_state_core.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your kubernetes_state_core performance data. See the [sample kubernetes_state_core.d/conf.yaml][4] for all available configuration options.

> 2. [Restart the Agent][5].

Reviewer: IMHO this is not ideal, it's not adapted to containerized environments in general and specifically KSM Core as it runs as a single instance in the cluster. Can we use the same approach we have on this page https://docs.datadoghq.com/integrations/kubernetes_state_core/?tab=helm#configuration ?

Author: Based on your remark, I think that the best is just to move all the content of https://docs.datadoghq.com/integrations/kubernetes_state_core/?tab=helm here, except the metrics list that would be in `metadata.csv`. The goal is to have the metrics list not only documented in the doc web site, but also in the app.
WDYT ?
-> https://github.com/DataDog/integrations-core/pull/12552/commits/1a426221cef82509ec6e9c49e3d62de20272db4c

Reviewer: Sounds good, thank you!

### Validation

[Run the Agent's status subcommand][6] and look for `kubernetes_state_core` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][7] for a list of metrics provided by this integration.

### Events

The Kubernetes State Core integration does not include any events.

### Service Checks

> See [service_checks.json][8] for a list of service checks provided by this integration.

Reviewer 3: It looks like service checks is empty. Should either add service checks in the json file, or change this line to "... does not include any service checks."

Author: It turns out that the service checks that are relying on the `kubernetes_state`-prefixed metrics described here are in the `kubernetes` directory: https://github.com/DataDog/integrations-core/blob/master/kubernetes/assets/service_checks.json

So, I’ve updated the links in the doc accordingly: https://github.com/DataDog/integrations-core/pull/12552/commits/713e17179bc06a1dac858cfd35a8ec928e40b220

## Troubleshooting

Need help? Contact [Datadog support][9].

> [1]: **LINK_TO_INTEGRATION_SITE**

Reviewer 2: 👀

Reviewer 3: ```suggestion
[1]: https://github.com/kubernetes/kube-state-metrics
```
